stdlib/random.ml

```diff
@@ -157,63 +136,43 @@ module State = struct
     then invalid_arg "Random.int64"
     else int64aux s bound
 
+  (* Return 32 or 64 random bits as a [nativeint] *)
+  let nativebits =
+    if Nativeint.size = 32
+    then fun s -> Nativeint.of_int32 (bits32 s)
+    else fun s -> Int64.to_nativeint (bits64 s)
 
+  (* Return a [nativeint] between 0 (included) and [bound] (excluded). *)
   let nativeint =
     if Nativeint.size = 32
     then fun s bound -> Nativeint.of_int32 (int32 s (Nativeint.to_int32 bound))
     else fun s bound -> Int64.to_nativeint (int64 s (Int64.of_nativeint bound))
 
+  (* Return a float 0 < x < 1 with at most 53 bits of precision. *)
+  let rec rawfloat s =
+    let b = next s in
+    let n = Int64.shift_right_logical b 11 in
+    if n <> 0L then Int64.to_float n *. 0x1.p-53 else rawfloat s
```

Comment:
I remember we had a discussion about how to draw floats recently, probably with @silene, and there was an explanation for why one approach was more desirable than the other. Could the reasons for using this one be mentioned in a comment?

Reply:
Since I was invoked in this thread, I will take the chance to point out that the comment is a truism. I guess it was meant to read "Return a float 0 < x < 1 that is a multiple of 2^-53".

Reply:
The discussion was in your RFC, https://github.com/ocaml/RFCs/pull/28#issuecomment-927270741

The bottom line for the discussion is that the current approach (multiples of 2^-53) is very simple and pretty good overall.

Agreed about the tautological nature of the comment.  (But at least it's not wrong!)  Will reword.

Reply:
By the way, since the only user of `rawfloat` is `float`, there is no point in excluding zero and thus to write a recursive function. Or is this an undocumented feature of `float` that, for normal values of `bound`, it returns a non-zero value?

Reply:
This was also discussed in the RFC: https://github.com/ocaml/RFCs/pull/28#issuecomment-927277230  Eventually, I'd like to export `rawfloat` (it's called `uniform` in PRINGO) with explicit guarantees that it returns a random FP between 0 and 1 excluded.